docs/make_api_index.py

```diff
@@ -0,0 +1,44 @@
+"""Generate the API index page for all ``movement`` modules."""
+
+import os
+
+# Modules to exclude from the API index
+exclude_modules = ["cli_entrypoint"]
+
+# Set the current working directory to the directory of this script
+script_dir = os.path.dirname(os.path.abspath(__file__))
+os.chdir(script_dir)
+
+
+def make_api_index():
```

Comment:
Is there any way we can define a custom ordering of modules, if we wish to ever override the default alphabetical ordering?

On one hand having a custom order would allow us to highlight the "important" modules near the top. But on the other hand, having that custom step will introduce manual work when creating/deleting a module, and the whole point of this PR was to eliminate manual work. So I think I'm fine with this staying alphabetical.

What do you think?